python/ray/serve/tests/test_standalone_3.py

```diff
@@ -672,5 +674,57 @@ def __call__(self):
     assert all_serve_logs.count("Deleting application 'default'") == 1
 
 
+def test_passing_object_ref_to_deployment_not_pinned_to_memory(
+    shutdown_ray, call_ray_stop_only  # noqa: F811
+):
+    """Test passing object ref to deployment not pinned to memory and cause memory leak.
+
+    We had issue that passing object ref to a deployment will result in memory leak
+    due to _PyObjScanner/ cloudpickler pinning the object to memory. This test will
+    ensure the object ref is release after the request is done.
+
+    See: https://github.com/ray-project/ray/issues/43248
+    """
+    cluster = Cluster()
+    cluster.add_node(num_cpus=5)
+    cluster.wait_for_nodes()
```

Comment:
Why do we need to use a cluster for this test?

Reply:
This kinda just following the same pattern as all other tests in this file😅 If no strong preference, would like to just keep as is

Reply:
I skimmed some of the other tests, and the only ones that really need a `cluster` here are those that use multiple proxies. In fact, I think you can run this test without using the Ray CLI at all. Can you move this test to `test_advanced.py`, remove the `cluster`, and use `serve_instance` as the only fixture?

Reply:
agreed no reason this needs to be standalone

Reply:
For reference, the `test_standalone` files should only contain tests that require starting an entirely new Ray cluster. We should try to minimize the tests we put in there because starting and stopping Ray clusters for each test is very slow.

Reply:
sg, I thought `test_standalone` was for anything that's not directly related to a module. Thanks for the explanation Shreyas!